### ISSUE TITLE
Error when $auth = null and phpredis = 5.0

### ISSUE THREAD

**Issue by nvahalik** [Namoshek/laravel-redis-sentinel#26]
We are in the process of updating from PHP 8.1 to 8.2. We upgraded from v0.1.2 to v0.4.0 and released. (Still on PHP 8.1). All of our production servers started reporting the following message:

    RedisException: Redis server 127.0.0.1:26379 went away.

From what I can tell the problem is this line here:

https://github.com/Namoshek/laravel-redis-sentinel/blob/6f9147845cf400ccc27de3d3c404062e42dc174d/src/Connectors/PhpRedisSentinelConnector.php#L124

In our case, `$auth = null`. If I change that line to the old version where `$auth` is not passed in, it works as expected. Adding it back in causes the error to come back.

**Comment by nvahalik** [comment]
I'm not 100% sure but the error seems to be here: 

https://github.com/phpredis/phpredis/blob/98d64ba86f37d2d3048500461f50b05f302f36ea/redis_sentinel.c#L100

```
    if (auth) {
        redis_sock_set_auth_zval(obj->sock, auth);
    }
```

This doesn't appear to be the correct way to handle `null` values? I am not 100% sure. I just know that this caused a major outage.

**Comment by Namoshek** [comment]
This must have been introduced by #25 where I refactored some of the old code. But I've done so with prior looking into the [documentation of phpredis](https://github.com/phpredis/phpredis/blob/release/5.3.6/sentinel.markdown) where it says the default is NULL, but I guess that doesn't mean you are allowed to pass NULL ... 🤦 

I'm going to fix this right away.

**Comment by Namoshek** [comment]
I've released a fix as [`v0.4.1`](https://github.com/Namoshek/laravel-redis-sentinel/releases/tag/v0.4.1).
